model_compression_toolkit/trainable_infrastructure/keras/quantize_wrapper.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2023 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from typing import Dict
+
+from model_compression_toolkit.constants import FOUND_TF
+from model_compression_toolkit.logger import Logger
+from model_compression_toolkit.trainable_infrastructure.common.base_trainable_quantizer import BaseTrainableQuantizer
+
+if FOUND_TF:
+    import tensorflow as tf
+    from mct_quantizers.keras.quantize_wrapper import KerasQuantizationWrapper as InferableKerasQuantizationWrapper
+
+    keras = tf.keras
+
+
+    def _weight_name(name: str) -> str:
+        """Extracts the weight name from the full TensorFlow variable name.
+
+        For example, returns 'kernel' for 'dense_2/kernel:0'.
+
+        Args:
+          name: TensorFlow variable name.
+
+        Returns:
+          Extracted weight name.
+        """
+        return name.split(':')[0].split('/')[-1]
+
+
+    class KerasTrainableQuantizationWrapper(InferableKerasQuantizationWrapper):
+
+        def _set_weights_vars(self, is_training: bool = True):
+            """
+            This function sets weights quantizers vars to the layer.
+            It's a duplicate of the function from InferableKerasQuantizationWrapper with a fix,
+            that should be fixed in MCT-quantizers wrapper. Once it's fixed there, this code
```

Comment:
Maybe open an issue for this on MCTQ and put a TODO comment here with a link to the relevant issue?

Reply:
A PR in MCTQ with this fix is waiting to be merged